### PR TITLE
Add profiler runs on regression

### PR DESCRIPTION
**Pull request by ddyurchenko**
### What does this PR do?

This PR adds profiler runs for benchmarks if regression was detected. This should add more confidence in results and allow to see what caused change in performance.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Changed code has unit tests for its functionality.
- [ ] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

File: .gitlab/scripts/run-benchmarks-with-profiler.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  -benchmem -count 10 -benchtime 2s ./...
+CANDIDATE_END_DATE=$(date -u --iso-8601=seconds | sed 's/\+.*/Z/')
+
+BASELINE_START_DATE=$(date -u --iso-8601=seconds | sed 's/\+.*/Z/')
+mkdir -p "${ARTIFACTS_DIR}/baseline-profile"
+cd "$BASELINE_SRC/ddtrace/tracer/"
+go test -run=XXX -bench "BenchmarkConcurrentTracing|BenchmarkStartSpan" \
+  -cpuprofile "${ARTIFACTS_DIR}/baseline-profile/cpu.pprof" \
+  -memprofile "${ARTIFACTS_DIR}/baseline-profile/delta-heap.pprof" \
+  -benchmem -count 10 -benchtime 2s ./...
```
Comment (felixge) [inline .gitlab/scripts/run-benchmarks-with-profiler.sh:26]: It seems like the list of benchmarks (and some of the options) are now duplicated in 4 places, (`run-benchmarks.sh` being the other file with 2 duplications).

Ideally that could be avoided (maybe via a shared bash include file). Alternatively some comment pointing out the duplications would be helpful.

File: .gitlab/scripts/run-benchmarks-with-profiler.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ "end":"$CANDIDATE_END_DATE",
+ "family":"go",
+ "version":"4"
+}
+END
+
+echo ""
+echo ""
+echo "Profiles were uploaded to Datadog! Open the following link to see them:"
+echo "https://dd.datad0g.com/profiling/comparison?query=service%3Add-trace-go-benchmarks&compare_end_A=$(date -d"$PROFILER_RUNS_END_DATE" +%s)000&compare_end_B=$(date -d"$PROFILER_RUNS_END_DATE" +%s)000&compare_query_A=service%3Add-trace-go-benchmarks%20config%3Acandidate%20%20job_id%3A${CI_JOB_ID}&compare_query_B=service%3Add-trace-go-benchmarks%20config%3Abaseline%20job_id%3A${CI_JOB_ID}&compare_start_A=$(date -d"$PROFILER_RUNS_START_DATE" +%s)000&compare_start_B=$(date -d"$PROFILER_RUNS_START_DATE" +%s)000&compareValuesMode=absolute&my_code=enabled&viz=flame_graph&paused=true"
```
Comment (felixge) [inline .gitlab/scripts/run-benchmarks-with-profiler.sh:75]: I think I'd prefer to target prod with this stuff, let's chat about it.

File: .gitlab/scripts/analyze-results.sh [hunk 3]
```diff
@@ -3,6 +3,7 @@
 # Change threshold for detection of regression
 # @see https://github.com/DataDog/relenv-benchmark-analyzer#what-is-a-significant-difference
 export UNCONFIDENCE_THRESHOLD=2.0
+export FAIL_ON_REGRESSION_THRESHOLD=2.0
```
Comment (felixge) [inline .gitlab/scripts/analyze-results.sh:6]: Is this an unrelated change?

**Review by felixge — APPROVED** [review]
LGTM, but the bash is getting a bit wild in terms of duplications. Extracting some stuff into a simple bash functions might help a lot with keeping the code readable and maintainable.

Not a blocker for merging this, but if you have some time ...

File: .gitlab/scripts/run-benchmarks-with-profiler.sh (hunk 2 above)
Comment (ddyurchenko) [reply .gitlab/scripts/run-benchmarks-with-profiler.sh:75]: It is easy change.

1. Need to generate DD_API_KEY in UI and add to AWS SSM under the name `ci.dd-trace-go.dd_api_key_prod` or similar, instead of the one `ci.dd-trace-go.dd_api_key` we use now. The instructions are the same like I have shown during our last sync.
2. Need to update url `https://intake.profile.datad0g.com/api/v2/profile` to production one.

Comment (felixge) [reply .gitlab/scripts/run-benchmarks-with-profiler.sh:75]: Yeah, this is not a blocker. We can do it later.

File: .gitlab/scripts/analyze-results.sh (hunk 3 above)
Comment (felixge) [reply .gitlab/scripts/analyze-results.sh:6]: 💡 Got it. Makes sense.

File: .gitlab/scripts/run-benchmarks-with-profiler.sh (hunk 1 above)
Comment (felixge) [reply .gitlab/scripts/run-benchmarks-with-profiler.sh:26]: Ah, the duplication is now actually a problem. This just landed: https://github.com/DataDog/dd-trace-go/pull/1672/files

This PR will need adjustment.

Comment (ddyurchenko) [reply .gitlab/scripts/run-benchmarks-with-profiler.sh:26]: I will update it :+1:

**Review by felixge — APPROVED** [review]
LGTM. 🚀
